vyper/signatures/interface.py

```diff
@@ -294,16 +294,14 @@ def check_valid_contract_interface(global_ctx, contract_sigs):
 
         for sig, func_sig in contract_sigs.items():
             if isinstance(func_sig, FunctionSignature):
-                # Remove units, as inteface signatures should not enforce units.
+                if sig not in funcs_left or func_sig.private:
+                    continue
```

Comment:
Can these two be separate `if` statements, with comments explaining why we ignore them?